### ISSUE TITLE
Instructions do not work - error in install

### ISSUE THREAD

**Issue by G0MJW** [funcube-dev/docker#7]
in ./build.sh 

pull access denied for fcbuild, repository does not exist or may require 'docker login'

**Comment by phooeydev** [comment]
Unfortunately, not a lot to go on, the message is exactly the same as in [this issue](https://github.com/funcube-dev/docker/issues/4)
However that is just a symptom of the fcbuild container not being successfully created (it is a prerequisite for the other containers). If its image cannot be found locally then docker will attempt to downloading from dockerhub, but obviously it doesn't exist there, hence the displayed error, I'll look at updating the script to make this clear.

To help diagnose could you post the output of the following commands:

`uname -a`
`docker run hello-world`
`docker images`
`docker ps -a`

and from the directory into which you cloned the docker repo:

`git status`
`./build.sh`

**Comment by G0MJW** [comment]
Yes – Docker is quite an obscure system and the error messages limited so when it goes wrong unless you are an expert with it it is hard to work out what is wrong.. That’s why I mostly stick to simple C.

 

uname -a

Linux raspberrypi 5.10.17+ #1414 Fri Apr 30 13:16:27 BST 2021 armv6l GNU/Linux

***@***.***:~ $ docker run hello-world

 

***@***.***:~ $ docker images

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE

golang              alpine3.12          e74bb656de39        4 days ago          280MB

hello-world         latest              851163c78e4a        17 months ago       4.85kB

***@***.***:~ $ docker ps -a

CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                       PORTS               NAMES

a67a96e56bdf        hello-world         "/hello"                 14 seconds ago      Exited (139) 4 seconds ago                       objective_clarke

c9f5cc88b6a7        e74bb656de39        "/bin/sh -c 'apk add…"   2 days ago          Exited (139) 2 days ago                          friendly_fermi

b0294ecdeb71        e74bb656de39        "/bin/sh -c 'apk add…"   2 days ago          Exited (139) 2 days ago                          compassionate_merkle

bad9fd7d742c        e74bb656de39        "/bin/sh -c 'apk add…"   2 days ago          Exited (139) 2 days ago                          dreamy_curran

a8f3f8da47db        e74bb656de39        "/bin/sh -c 'apk add…"   2 days ago          Exited (139) 2 days ago                          condescending_mccarthy

cb03120b293c        hello-world         "/hello"                 2 days ago          Exited (139) 2 days ago                          silly_ritchie

63d74835401a        hello-world         "/hello"                 2 days ago          Exited (139) 2 days ago                          determined_pascal

6618dfa6d0b4        e74bb656de39        "/bin/sh -c 'apk add…"   2 days ago          Exited (139) 2 days ago                          fervent_cartwright

18a1777be121        e74bb656de39        "/bin/sh -c 'apk add…"   2 days ago          Exited (139) 2 days ago                          pedantic_galois

6f8a2180dd68        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          modest_saha

d155a45b8aa9        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          vigilant_vaughan

02682411bff7        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          recursing_almeida

5c6336637e33        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          quizzical_brahmagupta

24881c8f5fbe        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          optimistic_morse

1bfe12ef7957        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          cocky_brahmagupta

5aabbfcff99d        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          vibrant_burnell

56616697d70c        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          friendly_heyrovsky

82c18221c9e4        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          silly_brattain

4b30c9920f6f        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          hardcore_mendel

53ac345176e4        e74bb656de39        "/bin/sh -c 'apk add…"   3 days ago          Exited (139) 3 days ago                          elegant_gauss

***@***.***:~ $

 

git status

On branch master

Your branch is up to date with 'origin/master'.

 

Changes not staged for commit:

  (use "git add <file>..." to update what will be committed)

  (use "git checkout -- <file>..." to discard changes in working directory)

 

        modified:   base-context/LimeSuite.tar.gz

        modified:   base-context/funcubeLib.tar.gz

        modified:   base-context/go.tar.gz

        modified:   update.sh

 

Untracked files:

  (use "git add <file>..." to include in what will be committed)

 

        LimeSuite/

        funcubeLib/

        go/

 

no changes added to commit (use "git add" and/or "git commit -a")

 

***@***.***:~/docker $ ./build.sh

Using path: /home/pi/docker

Tagging with: 1.0.5

Sending build context to Docker daemon  7.387MB

Step 1/15 : FROM golang:alpine3.12 AS build

---> e74bb656de39

Step 2/15 : RUN apk add     build-base     musl-dev     dpkg     libsndfile-dev                              portaudio-dev     libusb-dev     fftw-dev     boost-dev     cmake     py-mak                         o     sqlite     sqlite-dev     swig     py-six     git

---> Running in 34ab5feab4a3

The command '/bin/sh -c apk add     build-base     musl-dev     dpkg     libsndfile-dev     portaudio-dev     libusb-dev     fftw-dev     boost-dev     cmake     py-mako     sqlite     sqlite-dev     swig     py-six     git' returned a non-zero code: 139

Sending build context to Docker daemon  2.607kB

Step 1/9 : ARG FROM_TAG

Step 2/9 : FROM fcbuild:$FROM_TAG as fcbuild

pull access denied for fcbuild, repository does not exist or may require 'docker login'

Sending build context to Docker daemon  2.607kB

Step 1/10 : ARG FROM_TAG

Step 2/10 : FROM fcbuild:$FROM_TAG as fcbuild

pull access denied for fcbuild, repository does not exist or may require 'docker login'

Sending build context to Docker daemon  2.607kB

Step 1/10 : ARG FROM_TAG

Step 2/10 : FROM fcbuild:$FROM_TAG as fcbuild

pull access denied for fcbuild, repository does not exist or may require 'docker login'

Sending build context to Docker daemon  2.095kB

Step 1/8 : ARG FROM_TAG

Step 2/8 : FROM fcbuild:$FROM_TAG as fcbuild

pull access denied for fcbuild, repository does not exist or may require 'docker login'

Sending build context to Docker daemon  3.119kB

Step 1/12 : ARG FROM_TAG

Step 2/12 : FROM fcbuild:$FROM_TAG as fcbuild

pull access denied for fcbuild, repository does not exist or may require 'docker login'

***@***.***:~/docker $

**Comment by G0MJW** [comment]
Docker hello-world does not work… but 

 

docker run hypriot/armhf-hello-world

 

does work. Could this be because this is a PI2 ? I thought the decoder was compatible with the PI2 but maybe it isn’t any longer?

 

Mike
